File: BC-W4/Army/Spells/KissOfLight/KissOfLight.cpp
#include "KissOfLight.h"

KissOfLight::KissOfLight(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
KissOfLight::~KissOfLight() {};

void KissOfLight::action(Unit* enemy) {
    throw InvalidActionException();
};

void KissOfLight::heal(Unit* unit) {
    unit->addHitPoints(this->getActionPoints());
};

File: BC-W4/ArmyJava/src/spells/KissOfLight.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class KissOfLight extends Spell {
    public KissOfLight() {
            super("KissOfLight", 100, 120, false);
        }

    public KissOfLight(int actionPoints) {
        super("KissOfLight", 100, actionPoints, false);
    }

    public void heal(Unit unit) throws InvalidCastException {
        unit.addHitPoints(this.actionPoints);
    }
}
